File: problems-adapt/1401-1500/1490_deep-copy-an-n-ary-tree/solution_recursive.js
/**
 * @param {Node} root
 * @return {Node}
 */
var copyTree = function (root) {
    if (root === null) {
        return null;
    }
    const clone = new Node(root.val);
    for (const child of root.children) {
        clone.children.push(copyTree(child));
    }
    return clone;
};

File: problems-adapt/1401-1500/1490_deep-copy-an-n-ary-tree/solution_recursive.py
import sys
from typing import Optional


# Bundle-provided types (assembled with this submission):
#   Node: .val int, .children list[Node]


class Solution:
    def copyTree(self, root: Optional[Node]) -> Optional[Node]:
        # A 1000-deep chain nests 1000 calls, past CPython's default
        # recursion limit; lift it so the plain recursion stays viable.
        sys.setrecursionlimit(10000)
        if root is None:
            return None
        clone = Node(root.val)
        clone.children = [self.copyTree(child) for child in root.children]
        return clone
